File: code_samples/go/functional_closures.go
package main

import (
    "fmt"
)

// Functional: Closures
// AI/ML Training Sample

type Closures struct {
    Data string
}

func NewClosures() *Closures {
    return &Closures{
        Data: "",
    }
}

func (s *Closures) Process(input string) {
    s.Data = input
}

func (s *Closures) Validate() bool {
    return len(s.Data) > 0
}

func (s *Closures) GetData() string {
    return s.Data
}

func main() {
    instance := NewClosures()
    instance.Process("example")
    fmt.Printf("Data: %s\n", instance.GetData())
    fmt.Printf("Valid: %v\n", instance.Validate())
}

File: code_samples/scala/functional_closures.scala
/**
 * Functional: Closures
 * AI/ML Training Sample
 */

class Closures {
  private var data: String = ""
  
  def process(input: String): Unit = {
    data = input
  }
  
  def getData: String = data
  
  def validate: Boolean = data.nonEmpty
}

object ClosuresExample {
  def main(args: Array[String]): Unit = {
    val instance = new Closures()
    instance.process("example")
    println(s"Data: ${instance.getData}")
    println(s"Valid: ${instance.validate}")
  }
}
